src/Tasks/ResolveManifestFiles.cs

```diff
@@ -648,6 +665,15 @@ private ITaskItem GetOutputEntryPoint(ITaskItem entryPoint, List<PublishInfo> ma
 
         private bool IsFiltered(ITaskItem item)
         {
+            // In the case of .NET Core apps published as self-contained with loose files (i.e. PublishSingleFile != true),
+            // .NETCore binaries that come from the .NETCore Runtime pack should not be filtered out.
+            if (IsSelfContainedPublish && !IsSingleFilePublish &&
+                _runtimePackAssets != null &&
+                _runtimePackAssets.TryGetValue(item.ItemSpec, out _))
```

Comment:
Out of curiosity, why do we need to do filtering here, rather than letting the SDK's publish logic decide what to include?

Reply:
ClickOnce uses a different set of targets/tasks and has no dependency on SDK's publish logic since it works differently from SDK publish. The artifacts that ClickOnce publish generates are consumed by the ClickOnce runtime. These artifacts include manifests and bootstrapper exe's besides the published app's files.

ClickOnce publishing is available for both .NET FX and .NET Core apps and the same logic has to work for both types. The existing filtering logic is called during publish of both .NET FX apps and .NET Core. The current logic does not account for the SCD scenario since its a new scenario being added for ClickOnce.